cli-plugins/examples/helloworld/main.go

```diff
@@ -0,0 +1,25 @@
+package main
+
+import (
+	"fmt"
+
+	cliplugins "github.com/docker/cli/cli-plugins"
+	"github.com/docker/cli/cli-plugins/plugin"
+	"github.com/docker/cli/cli/command"
+	"github.com/spf13/cobra"
+)
+
+func main() {
+	plugin.Run(&plugin.Command{
+		Command: cobra.Command{
+			Use:   "helloworld",
+			Short: "A basic Hello World plugin for tests",
+		},
+		RunPlugin: func(cmd *cobra.Command, dockerCli command.Cli, args []string) {
+			fmt.Fprintln(dockerCli.Out(), "Hello World!")
+		},
```

Comment:
This model for plumbing `dockerCli` to the command runs is completely broken for subcommands... I'm going to have to rethink. I'm going to try having `plugin.Run` take a `makeRoot func(command.Cli) cobra.Command` in place of the `plugin.Command`, I think that can work.